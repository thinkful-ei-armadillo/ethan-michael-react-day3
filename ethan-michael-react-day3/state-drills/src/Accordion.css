ul {
  list-style-type: none;
  padding-left: 0;
}

li {
  border: 1px solid red;
  padding: 5px;
}
