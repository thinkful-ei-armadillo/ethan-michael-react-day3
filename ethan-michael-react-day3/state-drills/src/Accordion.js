import React, {Component} from 'react';

import './Accordion.css';

class Accordion extends Component{
    state = {
      openSectionID: null,
    };

    openSection = (id) => {
      this.setState({ openSectionID:  id });
    }

    createSections() {

      if (!this.props.sections || this.props.sections.length === 0) {
        return;
      }

      return this.props.sections.map((section, index) => {

        if (this.state.openSectionID === index) {

          // Open
          return (
            <li id={index} key={index}>
              <button onClick={() => { this.openSection(index) }} >{section.title}</button>
              <p>{section.content}</p>
            </li>
          );
        } else {

          // Closed
          return (
            <li id={index} key={index}>
              <button onClick={() => { this.openSection(index) }} >{section.title}</button>
            </li>
          );
        }
      });
    };

    render(){

        return (
          <ul>
            {this.createSections()}
          </ul>
        );
    }
}

export default Accordion;
