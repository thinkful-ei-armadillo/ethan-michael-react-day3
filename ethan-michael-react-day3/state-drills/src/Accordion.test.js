import React from 'react';
import ReactDOM from 'react-dom';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Accordion from './Accordion';

const testSections = [
  {
    title: 'Section 1',
    content: 'Lorem ipsum dolor sit amet consectetur adipisicing elit.',
  },
  {
    title: 'Section 2',
    content: 'Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!',
  },
  {
    title: 'Section 3',
    content: 'Animi amet cumque sint cupiditate officia ab voluptatibus libero optio et?',
  },
];

it('renders withou crashing', () => {

  const div = document.createElement('div');
  ReactDOM.render(<Accordion sections={testSections}/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('Snapshot Tests', () => {

  it('The component renders an empty ul when the sections prop is not supplied', () => {
    const wrapper = shallow(<Accordion />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('The component renders no sections as active by default', () => {
    const wrapper = shallow(<Accordion sections={testSections} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('The component opens a clicked section', () => {
    const wrapper = shallow(<Accordion sections={testSections} />);
    wrapper.find('button').at(0).simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('The component only opens the last section when multiple sections have been clicked.', () => {
    const wrapper = shallow(<Accordion sections={testSections} />);
    wrapper.find('button').at(0).simulate('click');
    wrapper.find('button').at(1).simulate('click');
    // wrapper.find('li#1 button').simulate('click');
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
