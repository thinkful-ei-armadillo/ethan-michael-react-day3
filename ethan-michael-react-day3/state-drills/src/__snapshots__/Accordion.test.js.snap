// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Tests The component only opens the last section when multiple sections have been clicked. 1`] = `
<ul>
  <li
    id={0}
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    id={1}
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
    <p>
      Cupiditate tenetur aliquam necessitatibus id distinctio quas nihil ipsam nisi modi!
    </p>
  </li>
  <li
    id={2}
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Snapshot Tests The component opens a clicked section 1`] = `
<ul>
  <li
    id={0}
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
    <p>
      Lorem ipsum dolor sit amet consectetur adipisicing elit.
    </p>
  </li>
  <li
    id={1}
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    id={2}
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;

exports[`Snapshot Tests The component renders an empty ul when the sections prop is not supplied 1`] = `<ul />`;

exports[`Snapshot Tests The component renders no sections as active by default 1`] = `
<ul>
  <li
    id={0}
    key="0"
  >
    <button
      onClick={[Function]}
    >
      Section 1
    </button>
  </li>
  <li
    id={1}
    key="1"
  >
    <button
      onClick={[Function]}
    >
      Section 2
    </button>
  </li>
  <li
    id={2}
    key="2"
  >
    <button
      onClick={[Function]}
    >
      Section 3
    </button>
  </li>
</ul>
`;
